Sources/SwiftProtobuf/JSONEncoder.swift

```diff
@@ -148,13 +173,19 @@ internal struct JSONEncoder {
                 append(staticText: "\"Infinity\"")
             }
         } else {
-            // TODO: Be smarter here about choosing significant digits
-            // See: protoc source has C++ code for this with interesting ideas
             if let v = Int64(safely: value) {
```

Comment:
With the special case for integers, we get `0` and `1`; without it, we get `0.0` and `1.0`.  Not significant, though the tests would need minor adjustments.

But:  With the special case for integers, `100 double` takes 19µs; without it, it takes 134µs.  For people who happen to be storing integer values in double fields, that's a big difference.